core/thread/group_spec.rb

```diff
@@ -1,5 +1,15 @@
 require_relative '../../spec_helper'
-require_relative 'fixtures/classes'
+
 describe "Thread#group" do
-  it "needs to be reviewed for spec completeness"
+  it "returns the default thread group for the main thread" do
+    Thread.main.group.should == ThreadGroup::Default
+  end
+
+  it "returns the thread group explicitly set for this thread" do
+    thread = Thread.new { nil }
+    thread_group = ThreadGroup.new
+    thread_group.add(thread)
+    thread.group.should == thread_group
+    thread.join
```

Comment:
```suggestion
  ensure
    thread.join
```
would be a bit safer, so if anything fails in between then the Thread isn't leaked

Reply:
Added the ensure block, and added a nil-check as well for the unlikely scenario that the thread object could not be created.